edx_django_utils/data_generation/tests/test_management.py

```diff
@@ -0,0 +1,501 @@
+"""
+Test management commands and related functions.
+"""
+
+import mock
+from django.core.management import get_commands, load_command_class
+from django.core.management.base import CommandError, SystemCheckError
+from django.test import TestCase
+from pytest import mark
+
+from edx_django_utils.data_generation.management.commands.manufacture_data import Command, Node
+# pylint: disable=unused-import
+from edx_django_utils.data_generation.tests.factories import (
+    TestCompanyFactory,
+    TestContactInfoFactory,
+    TestModelNonstandardCasingFactory,
+    TestNationalIdFactory,
+    TestPersonContactPhoneNumberFactory,
+    TestPersonFactory
+)
+from edx_django_utils.data_generation.tests.models import (
+    TestCompany,
+    TestContactInfo,
+    TestNationalId,
+    TestPerson,
+    TestPersonContactPhoneNumber,
+    test_model_nonstandard_casing
+)
+
+
+class TestCommand(Command):
+    """
+    Class for use in testing manufacture_data command via run_from_argv
+    """
+
+    def check(self, *args):
+        # Skip checks that aren't needed or configured in test suite
+        pass
+
+
+# Copied from django.core.management.__init__.py
+# https://github.com/django/django/blob/1ad7761ee616341295f36c80f78b86ff79d5b513/django/core/management/__init__.py#L83
+def call_command(command_name, *args, **options):
+    """
+    Call the given command, with the given options and args/kwargs.
+
+    This is the primary API you should use for calling specific commands.
+
+    `command_name` may be a string or a command object. Using a string is
+    preferred unless the command object is required for further processing or
+    testing.
+
+    Some examples:
+        call_command('migrate')
+        call_command('shell', plain=True)
+        call_command('sqlmigrate', 'myapp')
+
+        from django.core.management.commands import flush
+        cmd = flush.Command()
+        call_command(cmd, verbosity=0, interactive=False)
+        # Do something with cmd ...
+    """
+    app_name = get_commands()[command_name]
+    command = load_command_class(app_name, command_name)
+
+    # Simulate argument parsing to get the option defaults (see #10080 for details).
+    parser = command.create_parser("", command_name)
+    # Use the `dest` option name from the parser option
+    opt_mapping = {
+        min(s_opt.option_strings).lstrip("-").replace("-", "_"): s_opt.dest
+        for s_opt in parser._actions  # pylint: disable=protected-access
+        if s_opt.option_strings
+    }
+    arg_options = {opt_mapping.get(key, key): value for key, value in options.items()}
+    parse_args = []
+
+    defaults = parser.parse_args(args=parse_args)
+
+    # pylint: disable=protected-access
+    defaults = dict(
+        defaults._get_kwargs(), **arg_options
+    )
+    # Commented out section allows for unknown options to be passed to the command
```

Comment:
I feel like this should go to the top of the method, explaining why we have to redefine the core django function in the test.

Reply:
Went ahead and removed the commented out code, and explained the disabled arg checking at the top.  I figure if we ever want that code back, we know where we originally got it.